In [1]:
import pandas as pd

In [2]:
ls ./datasets

bom.movie_gross.csv*       imdb.title.ratings.csv*
imdb.name.basics.csv*      rt.movie_info.tsv*
imdb.title.akas.csv*       rt.reviews.tsv*
imdb.title.basics.csv*     tmdb.movies.csv*
imdb.title.crew.csv*       tn.movie_budgets.csv*
imdb.title.principals.csv*


# bom.movie_gross.csv

## columns title, studio, domestic_gross, foreign_gross, year

In [45]:
bom_df = pd.read_csv('datasets/bom.movie_gross.csv')

In [46]:
bom_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [47]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null object
year              3387 non-null int64
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [68]:
# remove the commas from the data in the foreign_gross column and convert it to float
bom_df['foreign_gross'] = bom_df['foreign_gross'].str.replace(',', '')

bom_df['foreign_gross'] = bom_df['foreign_gross'].astype('float')

In [69]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null float64
year              3387 non-null int64
dtypes: float64(2), int64(1), object(2)
memory usage: 132.4+ KB


why are there NAN in the foreign gross column?

is it because there is no foreign gross or because there is no data?

there are also some rows with no domestic gross, why?

In [70]:
# first lets look at the data with no foreign_gross
bom_df[bom_df['foreign_gross'].isna()]

title      studio  domestic_gross  \
222                                  Flipped          WB       1800000.0   
254   The Polar Express (IMAX re-issue 2010)          WB        673000.0   
267                           Tiny Furniture         IFC        392000.0   
269            Grease (Sing-a-Long re-issue)        Par.        366000.0   
280                          Last Train Home       Zeit.        288000.0   
...                                      ...         ...             ...   
3382                               The Quake       Magn.          6200.0   
3383             Edward II (2018 re-release)          FM          4800.0   
3384                                El Pacto        Sony          2500.0   
3385                                The Swan  Synergetic          2400.0   
3386                       An Actor Prepares       Grav.          1700.0   

      foreign_gross  year  
222             NaN  2010  
254             NaN  2010  
267             NaN  2010  
269             NaN  2010  
280             NaN  2010  
...             ...   ...  
3382            NaN  2018  
3383            NaN  2018  
3384            NaN  2018  
3385            NaN  2018  
3386            NaN  2018  

[1350 rows x 5 columns]

doing a google search on Flipped, I found that there is an international box office amount of $2,534,781 which is greater than the domestic_gross

listed at https://www.the-numbers.com/movie/Flipped#tab=summary

so foreign_gross info is available just not in this dataset

is it worth trying to scrape it? 1350 is 40 percent of the dataset, so it might be worth the effort

the data at the-numbers.com is not easily scraped and is in fact prohibited

In [73]:
# next lets look at the movies with no domestic_gross and sort them
for_sorted_bom_df = bom_df[bom_df['domestic_gross'].isna()].sort_values(by=['foreign_gross'], ascending=False)

In [74]:
for_sorted_bom_df

title   studio  domestic_gross  \
2825                       Secret Superstar      NaN             NaN   
2392                    Finding Mr. Right 2       CL             NaN   
1975         Surprise - Journey To The West       AR             NaN   
2468                                 Solace      LGP             NaN   
1268                             22 Bullets    Cdgm.             NaN   
871                            Solomon Kane     RTWC             NaN   
1659                             Jessabelle      LGF             NaN   
1308             Matru Ki Bijlee Ka Mandola      FIP             NaN   
928                            The Tall Man    Imag.             NaN   
537                                   Force     FoxS             NaN   
933                   Keith Lemon: The Film      NaN             NaN   
936                     Lula, Son of Brazil     NYer             NaN   
1681                              14 Blades     RTWC             NaN   
1685        Jack and the Cuckoo-Clock Heart   Shout!             NaN   
1340                      The Snitch Cartel       PI             NaN   
1342           All the Boys Love Mandy Lane     RTWC             NaN   
966                          The Cup (2012)     Myr.             NaN   
230              It's a Wonderful Afterlife      UTV             NaN   
1739                              Lila Lila    Crnth             NaN   
1368                                6 Souls     RTWC             NaN   
2595                                  Viral   W/Dim.             NaN   
1017                              Dark Tide      WHE             NaN   
298   Celine: Through the Eyes of the World     Sony             NaN   
302                              White Lion    Scre.             NaN   
1079                         The Green Wave       RF             NaN   
306                        Badmaash Company     Yash             NaN   
713                        Empire of Silver     NeoC             NaN   
327                      Aashayein (Wishes)  Relbig.             NaN   

      foreign_gross  year  
2825    122000000.0  2017  
2392    114700000.0  2016  
1975     49600000.0  2015  
2468     22400000.0  2016  
1268     21300000.0  2013  
871      19600000.0  2012  
1659      7000000.0  2014  
1308      6000000.0  2013  
928       5200000.0  2012  
537       4800000.0  2011  
933       4000000.0  2012  
936       3800000.0  2012  
1681      3800000.0  2014  
1685      3400000.0  2014  
1340      2100000.0  2013  
1342      1900000.0  2013  
966       1800000.0  2012  
230       1300000.0  2010  
1739      1100000.0  2014  
1368       852000.0  2013  
2595       552000.0  2016  
1017       432000.0  2012  
298        119000.0  2010  
302         99600.0  2010  
1079        70100.0  2012  
306         64400.0  2010  
713         19000.0  2011  
327          3800.0  2010

the highest grossing movie on this list, Secret Superstar, did not have a domestic_gross

and neither did the 2nd, Finding Mr. Right 2.

In [41]:
# column with 'international_gross' would be nice, but first I need to decide what to do with all the NaN values
bom_df['international_gross'] = bom_df['domestic_gross'] + bom_df['foreign_gross']

TypeError: unsupported operand type(s) for +: 'float' and 'str'